Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts

```diff
@@ -28,21 +29,45 @@ export class KeyVaultClient extends azureServiceClient.ServiceClient {
     }
 
     public async invokeRequest(request: webClient.WebRequest): Promise<webClient.WebResponse> {
-        try {
-            var response = await this.beginRequest(request);
-            if (response.statusCode == 401) {
-                var vaultResourceId = this.getValidVaultResourceId(response);
-                if(!!vaultResourceId) {
-                    console.log(tl.loc("RetryingWithVaultResourceIdFromResponse", vaultResourceId));
-                    
-                    this.getCredentials().activeDirectoryResourceId = vaultResourceId; // update vault resource Id
-                    this.getCredentials().getToken(true); // Refresh authorization token in cache
-                    var response = await this.beginRequest(request);
+        const maxRetryCount: number = 5;
```

Comment:
Doesn't the webclient take all these as parameters? https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/azure-arm-rest-v2/webClient.ts#L39

Reply:
We are not calling the web client directly. we are using '[beginRequest](https://github.com/microsoft/azure-pipelines-tasks/blob/4c3de8b30f2bf5a647d124254eae95a2e1f0698d/common-npm-packages/azure-arm-rest-v2/AzureServiceClientBase.ts#L105)' of azure service client which calls webclient.